File: exercise/java/day18/src/main/java/game/RockPaperScissors.java
package game;

public class RockPaperScissors {
    private RockPaperScissors() {
    }

    public static Result play(Choice player1, Choice player2) {
        if (player1 == player2) return new Result(Winner.DRAW, "same choice");
        else if (player1 == Choice.ROCK && player2 == Choice.SCISSORS)
            return new Result(Winner.PLAYER_1, "rock crushes scissors");
        else if (player1 == Choice.PAPER && player2 == Choice.ROCK)
            return new Result(Winner.PLAYER_1, "paper covers rock");
        else if (player1 == Choice.SCISSORS && player2 == Choice.PAPER)
            return new Result(Winner.PLAYER_1, "scissors cuts paper");
        else if (player2 == Choice.ROCK && player1 == Choice.SCISSORS)
            return new Result(Winner.PLAYER_2, "rock crushes scissors");
        else if (player2 == Choice.PAPER && player1 == Choice.ROCK)
            return new Result(Winner.PLAYER_2, "paper covers rock");
        else
            return new Result(Winner.PLAYER_2, "scissors cuts paper");
    }
}

File: exercise/php/day18/src/Game/RockPaperScissors.php
<?php

namespace Game;

class RockPaperScissors
{
    public static function play(string $player1, string $player2): array
    {
        if ($player1 === $player2) {
            return [
                'winner' => 'Draw',
                'reason' => 'same choice'
            ];
        } else if ($player1 === '🪨' && $player2 === '✂️') {
            return [
                'winner' => 'Player 1',
                'reason' => 'rock crushes scissors'
            ];
        } else if ($player1 === '📄' && $player2 === '🪨') {
            return [
                'winner' => 'Player 1',
                'reason' => 'paper covers rock'
            ];
        } else if ($player1 === '✂️' && $player2 === '📄') {
            return [
                'winner' => 'Player 1',
                'reason' => 'scissors cuts paper'
            ];
        } else if ($player2 === '🪨' && $player1 === '✂️') {
            return [
                'winner' => 'Player 2',
                'reason' => 'rock crushes scissors'
            ];
        } else if ($player2 === '📄' && $player1 === '🪨') {
            return [
                'winner' => 'Player 2',
                'reason' => 'paper covers rock'
            ];
        } else {
            return [
                'winner' => 'Player 2',
                'reason' => 'scissors cuts paper'
            ];
        }
    }
}
